x-pack/platform/plugins/shared/observability_ai_assistant/server/service/index.ts

```diff
@@ -90,7 +90,7 @@ export class ObservabilityAIAssistantService {
     ]);
 
     // user will not be found when executed from system connector context
-    const user = plugins.security.authc.getCurrentUser(request);
+    const user = plugins.security.authc.getCurrentUser(request)!;
```

Comment:
According to the comment above `user` is optional when called from system connector context so treating it as required could lead to null pointer exceptions.

Reply:
sorry yeah, quick workaround. will need to think of something else